sql/mysqld.cc

```diff
@@ -1000,7 +1002,7 @@ void deinit_rsa_keys(void);
 int show_rsa_public_key(THD *thd, SHOW_VAR *var, char *buff);
 #endif
 
-static volatile sig_atomic_t global_thread_count= 0;
+static std::atomic<uint> global_thread_count(0);
```

Comment:
Nit - you are not being consistent on using std::atomic<T> vs std::atomic_T (i.e. `thread_created` was `std::atomic_ulong` and `kill_blocked_pthreads_flag` was `std::atomic_bool` but this variable is `std::atomic<uint>`).  It's not a big deal - I just like consistency.  :)